Add checkstyle-enforced rule about bracing style (K&R style, braces are mandatory)

LukasKalbertodt opened the pull request:
Similar to #2216, but this time about bracing style. 

This follows a proposal by Greg on the mailing list in December 2019. Almost everyone agreed that we want K&R style braces (no newline before opening brace) and that we should make braces mandatory. There was no hard veto. 

As with the "120 line limit" rule a couple days ago, this rule is also disabled for a majority of modules as there are a lot of rule
violations. 

---

Here are the remaining modules with violations:

<details>
<summary>Sorted by name</summary>

```
admin-ui                                    => 175
adopter-statistics-impl                     => 5
annotation-impl                             => 16
asset-manager-impl                          => 16
asset-manager-storage-aws                   => 8
asset-manager-storage-fs                    => 3
asset-manager-workflowoperation             => 4
authorization-manager                       => 10
authorization-xacml                         => 2
caption-impl                                => 34
capture-admin-service-impl                  => 23
common                                      => 497
common-jpa-impl                             => 16
composer-ffmpeg                             => 139
composer-service-api                        => 16
composer-service-remote                     => 4
composer-workflowoperation                  => 99
conductor                                   => 5
cover-image-impl                            => 6
cover-image-workflowoperation               => 21
crop-api                                    => 2
db                                          => 9
distribution-service-aws-s3                 => 15
distribution-service-download               => 15
distribution-service-streaming-wowza        => 16
distribution-workflowoperation              => 25
dublincore                                  => 85
event-comment                               => 29
execute-impl                                => 19
execute-remote                              => 5
execute-workflowoperation                   => 18
external-api                                => 74
index-service                               => 188
ingest-download-service-impl                => 6
ingest-service-impl                         => 32
inspection-service-ffmpeg                   => 12
inspection-workflowoperation                => 3
kernel                                      => 53
list-providers-service                      => 13
live-schedule-impl                          => 35
lti-service-impl                            => 9
message-broker-api                          => 10
message-broker-impl                         => 2
metadata                                    => 4
metadata-api                                => 6
mpeg7                                       => 66
notification-workflowoperation              => 3
oaipmh                                      => 34
oaipmh-persistence                          => 17
publication-service-oaipmh                  => 38
publication-service-youtube-v3              => 4
scheduler-impl                              => 64
scheduler-remote                            => 10
schema                                      => 53
search                                      => 47
search-service-api                          => 9
search-service-impl                         => 191
search-service-remote                       => 6
security-aai                                => 8
security-shibboleth                         => 2
series-service-impl                         => 14
series-service-remote                       => 19
serviceregistry                             => 110
silencedetection-impl                       => 2
sox-impl                                    => 21
sox-workflowoperation                       => 9
static                                      => 3
static-file-service-api                     => 2
static-file-service-impl                    => 3
textanalyzer-impl                           => 4
textanalyzer-workflowoperation              => 12
textextractor-tesseract                     => 3
themes                                      => 7
themes-workflowoperation                    => 2
timelinepreviews-ffmpeg                     => 7
transcription-service-ibm-watson-impl       => 22
transcription-service-workflowoperation     => 14
urlsigning-service-impl                     => 4
user-interface-configuration                => 2
userdirectory                               => 74
userdirectory-ldap                          => 31
userdirectory-moodle                        => 30
userdirectory-sakai                         => 6
usertracking-impl                           => 28
videoeditor-ffmpeg-impl                     => 9
videoeditor-workflowoperation               => 7
videogrid-service-impl                      => 3
videogrid-workflowoperation                 => 16
videosegmenter-ffmpeg                       => 5
waveform-ffmpeg                             => 2
workflow-condition-parser                   => 5
workflow-service-api                        => 82
workflow-service-impl                       => 49
workflow-service-remote                     => 25
workflow-workflowoperation                  => 18
working-file-repository-service-impl        => 29
working-file-repository-service-remote      => 4
workspace-impl                              => 9
```
</details>

<details>
<summary>Sorted by number of violations (not number of files with violations)</summary>

```
authorization-xacml                         => 2
crop-api                                    => 2
message-broker-impl                         => 2
security-shibboleth                         => 2
silencedetection-impl                       => 2
static-file-service-api                     => 2
themes-workflowoperation                    => 2
user-interface-configuration                => 2
waveform-ffmpeg                             => 2
asset-manager-storage-fs                    => 3
inspection-workflowoperation                => 3
notification-workflowoperation              => 3
static                                      => 3
static-file-service-impl                    => 3
textextractor-tesseract                     => 3
videogrid-service-impl                      => 3
asset-manager-workflowoperation             => 4
composer-service-remote                     => 4
metadata                                    => 4
publication-service-youtube-v3              => 4
textanalyzer-impl                           => 4
urlsigning-service-impl                     => 4
working-file-repository-service-remote      => 4
adopter-statistics-impl                     => 5
conductor                                   => 5
execute-remote                              => 5
videosegmenter-ffmpeg                       => 5
workflow-condition-parser                   => 5
cover-image-impl                            => 6
ingest-download-service-impl                => 6
metadata-api                                => 6
search-service-remote                       => 6
userdirectory-sakai                         => 6
themes                                      => 7
timelinepreviews-ffmpeg                     => 7
videoeditor-workflowoperation               => 7
asset-manager-storage-aws                   => 8
security-aai                                => 8
db                                          => 9
lti-service-impl                            => 9
search-service-api                          => 9
sox-workflowoperation                       => 9
videoeditor-ffmpeg-impl                     => 9
workspace-impl                              => 9
authorization-manager                       => 10
message-broker-api                          => 10
scheduler-remote                            => 10
inspection-service-ffmpeg                   => 12
textanalyzer-workflowoperation              => 12
list-providers-service                      => 13
series-service-impl                         => 14
transcription-service-workflowoperation     => 14
distribution-service-aws-s3                 => 15
distribution-service-download               => 15
annotation-impl                             => 16
asset-manager-impl                          => 16
common-jpa-impl                             => 16
composer-service-api                        => 16
distribution-service-streaming-wowza        => 16
videogrid-workflowoperation                 => 16
oaipmh-persistence                          => 17
execute-workflowoperation                   => 18
workflow-workflowoperation                  => 18
execute-impl                                => 19
series-service-remote                       => 19
cover-image-workflowoperation               => 21
sox-impl                                    => 21
transcription-service-ibm-watson-impl       => 22
capture-admin-service-impl                  => 23
distribution-workflowoperation              => 25
workflow-service-remote                     => 25
usertracking-impl                           => 28
event-comment                               => 29
working-file-repository-service-impl        => 29
userdirectory-moodle                        => 30
userdirectory-ldap                          => 31
ingest-service-impl                         => 32
caption-impl                                => 34
oaipmh                                      => 34
live-schedule-impl                          => 35
publication-service-oaipmh                  => 38
search                                      => 47
workflow-service-impl                       => 49
kernel                                      => 53
schema                                      => 53
scheduler-impl                              => 64
mpeg7                                       => 66
external-api                                => 74
userdirectory                               => 74
workflow-service-api                        => 82
dublincore                                  => 85
composer-workflowoperation                  => 99
serviceregistry                             => 110
composer-ffmpeg                             => 139
admin-ui                                    => 175
index-service                               => 188
search-service-impl                         => 191
common                                      => 497
```
</details>


---

### Your pull request should…

* [ ] have a concise title
* [ ] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [ ] be against the correct branch (features can only go into develop)
* [ ] include migration scripts and documentation, if appropriate
* [ ] pass automated tests
* [ ] have a clean commit history
* [ ] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

LukasKalbertodt committed:
Add checkstyle rule to enforce consistent bracing style

This follows a proposal by Greg on the mailing list in December 2019.
Almost everyone agreed that we want K&R style braces (no newline before
opening brace) and that we should make braces mandatory. There was no
hard veto.

As with the "120 line limit" rule a couple days ago, this rule is also
disabled for a majority of modules as there are a lot of rule
violations. We expect to correctly format modules one by one and then
remove those style exceptions.

LukasKalbertodt committed:
Fix brace style for modules with only one one such style violation

lkiesow commented:
I don't think autoformatting will break this (hence far less important) but could we add this to the IDEA/Eclipse auto-formatting rules?

LukasKalbertodt committed:
Adjust IntelliJ codestyle to new brace style

The default position of braces already matches our style, but this now
also enforces braces in various contexts.

LukasKalbertodt commented:
@lkiesow I pushed a commit that adjusts the config for IntelliJ. 

I don't have Eclipse right now and I never worked with it. Also, from reading docs about the XML format, I can't find any option that would enforce braces. The brace position is already correct in the Eclipse style. And as you said, the formatter won't produce code incompatible with checkstyle, so it's not that crucial, I guess.

LukasKalbertodt committed:
Add Eclipse code cleanup template

This file was sent to my by Greg. I couldn't test it myself as I have
never used Eclipse.
